Source/WebCore/css/StyleColor.h

```diff
@@ -110,25 +115,107 @@ class StyleColor {
     static bool isColorKeyword(CSSValueID, OptionSet<CSSColorType> = { CSSColorType::Absolute, CSSColorType::Current, CSSColorType::System });
 
     bool isCurrentColor() const;
+    bool isColorMix() const;
     bool isAbsoluteColor() const;
     const Color& absoluteColor() const;
 
     WEBCORE_EXPORT Color resolveColor(const Color& colorPropertyValue) const;
 
+    friend String serializationForCSS(const StyleColor&);
+    friend void serializationForCSS(StringBuilder&, const StyleColor&);
     friend WTF::TextStream& operator<<(WTF::TextStream&, const StyleColor&);
     String debugDescription() const;
 
 private:
-    using ColorKind = std::variant<Color, CurrentColor>;
+    using ColorKind = std::variant<Color, StyleCurrentColor, UniqueRef<StyleColorMix>>;
 
-    StyleColor(const ColorKind& color)
-        : m_color { color }
+    StyleColor(ColorKind&& color)
+        : m_color { WTFMove(color) }
     {
     }
 
+    static ColorKind resolveAbsoluteComponents(StyleColorMix&&);
+    static ColorKind copy(const ColorKind&);
+
     ColorKind m_color;
 };
 
+struct StyleColorMix {
+    WTF_MAKE_FAST_ALLOCATED;
```

Comment:
There's a flavor of this macro for structs that you can use so you don't have to explicitly say `public:` after it.

Reply:
Oh nice. I didn't think to look. Thanks.